### PR TITLE
Fix - Transformers have only one substation

### PR DESCRIPTION
**Pull request by basseche**

### CODE REVIEW

File: src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java [hunk 1]
```diff
@@ -383,6 +382,7 @@ private static String getThreeWindingsTransformerFieldValue(FieldType field, Str
         return switch (field) {
             case COUNTRY -> threeWindingsTransformer.getSubstation().flatMap(Substation::getCountry).map(String::valueOf).orElse(null);
             case RATED_VOLTAGE_0 -> String.valueOf(threeWindingsTransformer.getRatedU0());
+            case SUBSTATION_PROPERTIES -> threeWindingsTransformer.getNullableSubstation().getProperty(propertyName);
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java:385]: we should add an error if the substation is null

Comment (EtienneLt) [reply src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java:385]: or juste return null if the substation is null

**Comment by EtienneLt** [comment]
I'm asking @FranckLecuyer  but I think we can do something in FiltersUtils too

File: src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java [hunk 2]
```diff
@@ -371,8 +371,8 @@ private static String getTwoWindingsTransformerFieldValue(FieldType field, Strin
             case HAS_PHASE_TAP_CHANGER -> String.valueOf(twoWindingsTransformer.hasPhaseTapChanger());
             case PHASE_REGULATION_MODE,
                 PHASE_REGULATION_VALUE -> getPhaseTapChangerFieldValue(field, twoWindingsTransformer.getPhaseTapChanger());
-            case SUBSTATION_PROPERTIES_1 -> twoWindingsTransformer.getTerminal1().getVoltageLevel().getNullableSubstation().getProperty(propertyName);
-            case SUBSTATION_PROPERTIES_2 -> twoWindingsTransformer.getTerminal2().getVoltageLevel().getNullableSubstation().getProperty(propertyName);
+            case SUBSTATION_PROPERTIES -> twoWindingsTransformer.getNullableSubstation() != null ?
+                twoWindingsTransformer.getNullableSubstation().getProperty(propertyName) : null;
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java:375]: could you add a test when tje substation is null ?

**Review by EtienneLt — COMMENTED** [review]
just to pass sonar and it is good

File: src/main/java/org/gridsuite/filter/utils/expertfilter/FieldType.java [hunk 3]
```diff
@@ -115,7 +115,6 @@ public enum FieldType {
     SUBSTATION_PROPERTIES,
     SUBSTATION_PROPERTIES_1,
     SUBSTATION_PROPERTIES_2,
-    SUBSTATION_PROPERTIES_3,
```
Comment (Mathieu-Deharbe) [inline src/main/java/org/gridsuite/filter/utils/expertfilter/FieldType.java:118]: Even if this value is obsolete we might have to keep it (with a comment explaining why) because otherwise it is now impossible to get inside a folder containing a filter using SUBSTATION_PROPERTIES_3. Because of the following crash on filter-server : 

```
java.lang.IllegalArgumentException: No enum constant org.gridsuite.filter.utils.expertfilter.FieldType.SUBSTATION_PROPERTIES_3
	at java.base/java.lang.Enum.valueOf(Enum.java:273) ~[na:na]
	at org.hibernate.type.descriptor.java.EnumJavaType.fromName(EnumJavaType.java:254) ~[hibernate-core-6.5.2.Final.jar:6.5.2.Final]
```

File: src/main/java/org/gridsuite/filter/utils/expertfilter/FieldType.java [hunk 4]
```diff
@@ -115,7 +115,6 @@ public enum FieldType {
     SUBSTATION_PROPERTIES,
     SUBSTATION_PROPERTIES_1,
     SUBSTATION_PROPERTIES_2,
```
Comment (Mathieu-Deharbe) [inline src/main/java/org/gridsuite/filter/utils/expertfilter/FieldType.java:117]:
```suggestion
    SUBSTATION_PROPERTIES_2,
    SUBSTATION_PROPERTIES_3, // this FieldType value is obsolete but kept here to avoid crashes in case of remaining filters using it
```

File: src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java (hunk 2 above)
Comment (basseche) [reply src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java:375]: done

**Review by EtienneLt — APPROVED** [review]
